### DISCUSSION TITLE
Add configurable keyboard shortcuts (e.g. shortcut for switching shown display)

### DISCUSSION THREAD

**Discussion by Jackenmen** [rustdesk/rustdesk#1933] [Q&A] [answered]
It would be great if it were possible to configure keyboard shortcuts in the RustDesk client which could be used while in a session.

In particular, I would like to be able to set a keyboard shortcut for:
- switching the current display
    - there are two ways this could be done, either there could be shortcuts for selecting a display with a specific number, *or* there could be shortcuts for going to the previous/next display. Personally, I prefer the latter but I would be fine with the former as well. You could also allow both.
- toggling full-screen mode

I imagine there are more shortcuts that could be useful, one that comes to my mind is sending Ctrl+Alt+Delete.

I don't know if there should be defaults set for these but either way, it should be possible to disable any shortcut, no matter if it's a default one or a configured one.

**Comment by qt06** [comment]
That's very useful. Especially for me.
Because of my eyes, I can't use the mouse. I just use the screen reader software  that help me to read the screen.
So, I need a shortcut to toggle local and remote.

**Comment by alzi123** [comment]
You can use this autohotkey script in activated full screen mode:
You have to change the coordinates of course.
The window of RustDesk has to be activated and the shortcut used is
Shift+Strg+1 for the first monitor
Shift+Strg+2 for the second monitor
Shift+Strg+3 for the third monitor
Shift+Strg+4 for the fourth monitors
at the remote computer.
What a difference in the workflow !!
Have Fun.


^+1::
CoordMode, Mouse, Screen
MouseGetPos xpos, ypos 
MouseMove 2490,-510
Sleep, 300
MouseMove 2490,-495
Click
MouseMove, xpos, ypos, 0
Return

^+2::
CoordMode, Mouse, Screen
MouseGetPos xpos, ypos 
MouseMove 2520,-510
Sleep, 300
MouseMove 2520,-495
Click
MouseMove, xpos, ypos, 0
Return

^+3::
CoordMode, Mouse, Screen
MouseGetPos xpos, ypos 
MouseMove 2545,-510
Sleep, 300
MouseMove 2545,-495
Click
MouseMove, xpos, ypos, 0
Return

^+4::
CoordMode, Mouse, Screen
MouseGetPos xpos, ypos 
MouseMove 2575,-510
Sleep, 300
MouseMove 2575,-495
Click
MouseMove, xpos, ypos, 0
Return

**Reply by open-trade** [reply]
Yes it would be very practical. Google Chrome Desktop has such functionnality. You can set a special key (for an example, I've affected "²", that I never use) and after that, you can do some actions by pressing ²
![image](https://github.com/rustdesk/rustdesk/assets/3479885/e9a661ec-275b-4ea1-a6f4-f59318a78e28)

- toggle full screen
- change screen

and so on...

**Comment by adrianog** [comment]
am I right in understanding that there is no keyboard shortcut whatsoever?
e.g. to change the input source?

**Comment by mbtolou** [comment]
![image](https://github.com/user-attachments/assets/3cacf688-068e-4f4e-a432-dad28eaee66c)

Thank you for your great software.
I often have to change the positions of the monitors dozens of times a day, and it’s very annoying.
Could you add the monitors to the small ToolBar section shown in the picture?

**Reply by zacaj** [reply]
this would be super handy, but probably a different feature request?

**Reply by mbtolou** [reply]
This is definitely a different request than this feature, can you develop it in the software?

**Reply by zacaj** [reply]
I made a separate request for it: https://github.com/rustdesk/rustdesk/discussions/12399

**Reply by mbtolou** [reply]
thanks

**Comment by rustdesk** [comment, marked as answer]
Here is [a draft implementation](https://github.com/rustdesk/rustdesk/pull/14933). We are looking for a contributor to [test](https://github.com/rustdesk/rustdesk/pull/14933#issuecomment-4351056842) and polish it.
